sympy/stats/stochastic_process_types.py

```diff
@@ -1335,13 +1336,13 @@ def probability(self, condition, given_condition=None, evaluate=True, **kwargs):
             if check_numeric and check_given_numeric:
                 res = []
                 if isinstance(condition, Or):
-                    res.append(Add(*[self._solve_numerical(arg, given_condition)
+                    res.append(Add.fromiter([self._solve_numerical(arg, given_condition)
                             for arg in condition.args]))
```

Comment:
Have you tried `Add.fromiter(self._solve_numerical(arg, given_condition) for arg in condition.args)` without square brackets? I think it's a bit more efficient because it creates a generator instead of a list, so no need to allocate all the memory for the array.

Reply:
Yes, it works. Thanks!!